src/main/kotlin/com/colivery/serviceaping/rest/v1/services/HelpSeekerRestService.kt

```diff
@@ -16,34 +16,64 @@ import org.springframework.validation.Errors
 import org.springframework.validation.annotation.Validated
 import org.springframework.web.bind.annotation.*
 import reactor.core.publisher.Mono
+import java.util.*
 
 @RestController
 @Validated
 @RequestMapping("/v1/help-seeker", produces = [MediaType.APPLICATION_JSON_VALUE])
 class HelpSeekerRestService(
-        private val helpSeekerRepository: HelpSeekerRepository
+    private val helpSeekerRepository: HelpSeekerRepository
 ) {
 
     @PostMapping
-    fun createHelpSeeker(@RequestBody helpSeeker: CreateHelpSeekerDto, authentication: Authentication):
+    fun createHelpSeeker(
+        @RequestBody helpSeeker: CreateHelpSeekerDto,
+        authentication: Authentication
+    ):
             Mono<HelpSeekerResource> {
         val errors: Errors = BeanPropertyBindingResult(helpSeeker, "help_seeker")
 
         if (errors.hasErrors()) {
             throw BadRequestException()
         }
 
-        val helpSeekerEntity = this.helpSeekerRepository.save(toHelpSeekerEntity(helpSeeker, authentication.getUser()))
+        val helpSeekerEntity =
+            this.helpSeekerRepository.save(toHelpSeekerEntity(helpSeeker, authentication.getUser()))
 
         val resource = toHelpSeekerResource(helpSeekerEntity)
         return Mono.just(resource)
     }
 
+    @PutMapping
+    @PreAuthorize("hasRole('ROLE_HOTLINE')")
+    fun updateHelpSeeker(
+        @PathVariable helpSeekerId: UUID, @RequestBody helpSeekerData:
```

Comment:
just for better readability, maybe move `@RequestBody helpSeekerData:` into the next line like in the upper method
```suggestion
        @PathVariable helpSeekerId: UUID,
        @RequestBody helpSeekerData: CreateHelpSeekerDto
```